include/sparrow/layout/mutable_array_base.hpp

```diff
@@ -88,10 +91,54 @@ namespace sparrow
         iterator insert(const_iterator pos, const value_type& value);
         iterator insert(const_iterator pos, const value_type& value, size_type count);
         iterator insert(const_iterator pos, std::initializer_list<value_type> values);
-        template <typename InputIt>
-        iterator insert(const_iterator pos, InputIt first, InputIt last);
+
+        template <mpl::iterator_of_type<value_type> InputIt>
+        iterator insert(const_iterator pos, InputIt first, InputIt last)
```

Comment:
I just changed the template by using `mpl::iterator_of_type<value_type>` instead of `typename`
But I fail to separate the definition and implementation, I get this error:
> out-of-line definition of 'insert' does not match any declaration in 'mutable_array_base<D>'

Reply:
I had the same issue when I tried to move some primitive_array construtor definitions out of the class. Even if it would be nicer to have these definitions outside of the class, let's not waste time on it and see later if wa can find a solution for that.

Reply:
Unfortunately no. I just put the implementation with the declaration